proto/spacemesh/types.proto

```diff
@@ -103,28 +103,54 @@ message TransactionState {
     TransactionStateType state = 2;
 }
 
+// Data specific to an activation transaction.
+message ActivationTransactionData {
+    SmesherId smesher_id = 1;    // used in atx only
+    TransactionId prev_atx = 2;  // previous ATX. used in atx.
+    uint64 commitment_size = 3; // commitment size in bytes
+}
+
+// Data specific to a simple coin transaction.
+message CoinTransactionData {
+    AccountId receiver = 1;
+}
+
+message SmartContractTransactionData {
+    enum TransactionType {
+        APP = 0;             // smart contract method
+        SPAWN_APP = 1;       // deploy app from template using svm terminology
+        DEPLOY_TEMPLATE = 2; // deploy app template code to mesh
+    }
+    TransactionType type = 1;
+    bytes data = 2;          // packed binary arguments
+}
+
+// A simple signature data
+message Signature {
+    enum Scheme {
+        ED25519 = 0;            // Standard Ed25519 scheme
+        ED25519_PLUS_PLUS = 1;  // Spacemesh-modified ED25519 scheme, a.k.a. ED25519++
+    }
+    Scheme scheme = 1;    // The signature's scheme
+    bytes signature = 2;  // The signature itself
+    bytes public_key = 3; // Included in schemes which require signer to provide a public key
```

Comment:
How do we link this to AccountId/address? Do we want to do that here, or just leave them distinct?

Reply:
I don't understand the question. what is this?

Reply:
Public key is not the same thing as AccountId - how do we cross-reference the two?

Reply:
I think that the public key is only interesting in the context of a signature that includes it. e.g. ED25519 as this data will be on the mesh. It is a good question regarding the need to include account id in coin transactions that use this scheme as it can be extracted from the public key...

Reply:
Maybe we create it from the public key and it doesn't need to be included in the tx itself but when we create a transaction data item we populate it with the account id for convenience

Reply:
I think we should live then distinct even though the account is the 20 bytes out of the key.

Reply:
I and I think they are cross-referenced by the fact the Signature is a member of Transaction always has an AccountId for sender.

Reply:
I don't think we'd ever want to show the signature, or the public key, inside the downstream, user-facing apps such as wallet/explorer. We'll want AccountId (address) instead. That's why I'm asking. Do we want the backend to convert from public key -> address, and include this explicitly, or do we expect the downstream apps to do the public key conversion/extraction themselves?

Reply:
there's another field for the account so they don't need to extract. I think it is useful to include the signature in transactions but I don't feel strongly about it. It will be useful to do analysis regarding what signature scheme used and for which transaction. It can be extracted from txs and the mesh by any full node anyhow so this is just convenience.

Reply:
ok